File: fantasy-chess-frontend/test-discord-env.js
#!/usr/bin/env node

/**
 * Discord Environment Variables Test
 * Tests what environment variables the function can access
 */

import https from 'https';
import { readFileSync } from 'fs';
import { fileURLToPath } from 'url';
import { dirname, join } from 'path';

// Get the directory of the current file
const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

// Load environment variables from .env file
function loadEnv() {
  try {
    const envPath = join(__dirname, '.env');
    const envContent = readFileSync(envPath, 'utf8');
    const env = {};
    
    envContent.split('\n').forEach(line => {
      const [key, ...valueParts] = line.split('=');
      if (key && valueParts.length > 0) {
        env[key.trim()] = valueParts.join('=').trim();
      }
    });
    
    return env;
  } catch (error) {
    console.log('⚠️ Could not load .env file');
    return {};
  }
}

const env = loadEnv();

// Helper function to make HTTPS requests
function makeRequest(options, data) {
  return new Promise((resolve, reject) => {
    const req = https.request(options, (res) => {
      let responseData = '';
      
      res.on('data', (chunk) => {
        responseData += chunk;
      });
      
      res.on('end', () => {
        resolve({
          statusCode: res.statusCode,
          headers: res.headers,
          data: responseData
        });
      });
    });
    
    req.on('error', (error) => {
      reject(error);
    });
    
    if (data) {
      req.write(data);
    }
    
    req.end();
  });
}

// Test 1: Check environment variables
async function testEnvironmentVariables() {
  console.log('🔍 Test 1: Check environment variables...');
  
  const requestData = JSON.stringify({
    action: 'create_league_channel',
    leagueName: 'Test League',
    leagueId: 'test-league-env'
  });
  
  const options = {
    hostname: 'wdbwzvnkfbyzazodfhsw.supabase.co',
    port: 443,
    path: '/functions/v1/discord-bot',
    method: 'POST',
    headers: {
      'Authorization': `Bearer ${env.SUPABASE_SERVICE_ROLE_KEY}`,
      'Content-Type': 'application/json',
      'Content-Length': Buffer.byteLength(requestData)
    }
  };
  
  try {
    const response = await makeRequest(options, requestData);
    console.log(`Status: ${response.statusCode}`);
    console.log(`Response: ${response.data}`);
    
    // Parse the response to see the error details
    try {
      const errorData = JSON.parse(response.data);
      if (errorData.details) {
        console.log('\n🔍 Error details:', errorData.details);
      }
      if (errorData.suggestion) {
        console.log('💡 Suggestion:', errorData.suggestion);
      }
    } catch (e) {
      // Could not parse as JSON
    }
    
    return response;
  } catch (error) {
    console.log('❌ Error:', error.message);
    return null;
  }
}

// Test 2: Check Discord API directly
async function testDiscordApi() {
  console.log('\n🔍 Test 2: Check Discord API directly...');
  
  if (!env.DISCORD_BOT_TOKEN) {
    console.log('❌ DISCORD_BOT_TOKEN not found in .env file');
    return;
  }
  
  if (!env.DISCORD_MAIN_SERVER_ID) {
    console.log('❌ DISCORD_MAIN_SERVER_ID not found in .env file');
    return;
  }
  
  console.log('✅ Found Discord credentials in .env file');
  console.log(`Bot Token: ${env.DISCORD_BOT_TOKEN.substring(0, 10)}...`);
  console.log(`Server ID: ${env.DISCORD_MAIN_SERVER_ID}`);
  
  // Test Discord API directly
  const options = {
    hostname: 'discord.com',
    port: 443,
    path: `/api/v10/guilds/${env.DISCORD_MAIN_SERVER_ID}`,
    method: 'GET',
    headers: {
      'Authorization': `Bot ${env.DISCORD_BOT_TOKEN}`,
      'Content-Type': 'application/json'
    }
  };
  
  try {
    const response = await makeRequest(options);
    console.log(`Discord API Status: ${response.statusCode}`);
    console.log(`Discord API Response: ${response.data}`);
    
    if (response.statusCode === 200) {
      console.log('✅ Discord API is accessible!');
    } else {
      console.log('❌ Discord API returned error');
    }
  } catch (error) {
    console.log('❌ Discord API error:', error.message);
  }
}

// Test 3: Check Supabase environment variables
async function checkSupabaseEnv() {
  console.log('\n🔍 Test 3: Check what Supabase environment variables are set...');
  
  console.log('Local .env file contains:');
  console.log(`- SUPABASE_URL: ${env.SUPABASE_URL ? '✅ Set' : '❌ Missing'}`);
  console.log(`- SUPABASE_SERVICE_ROLE_KEY: ${env.SUPABASE_SERVICE_ROLE_KEY ? '✅ Set' : '❌ Missing'}`);
  console.log(`- DISCORD_BOT_TOKEN: ${env.DISCORD_BOT_TOKEN ? '✅ Set' : '❌ Missing'}`);
  console.log(`- DISCORD_MAIN_SERVER_ID: ${env.DISCORD_MAIN_SERVER_ID ? '✅ Set' : '❌ Missing'}`);
  
  console.log('\n💡 Note: The function uses Supabase Dashboard environment variables, not local .env file');
  console.log('Make sure these are set in Supabase Dashboard → Settings → Environment Variables:');
  console.log('- DISCORD_BOT_TOKEN');
  console.log('- DISCORD_MAIN_SERVER_ID');
}

// Main test runner
async function runEnvTests() {
  console.log('🔧 Discord Environment Variables Test');
  console.log('=' .repeat(50));
  console.log('');
  
  await testEnvironmentVariables();
  await testDiscordApi();
  await checkSupabaseEnv();
  
  console.log('\n' + '=' .repeat(50));
  console.log('🔧 Troubleshooting:');
  console.log('1. Check Supabase Dashboard → Settings → Environment Variables');
  console.log('2. Make sure DISCORD_BOT_TOKEN and DISCORD_MAIN_SERVER_ID are set');
  console.log('3. Verify the bot token is valid and the server ID is correct');
  console.log('4. Check if the bot is added to the server with proper permissions');
  console.log('5. Look at Supabase Dashboard → Edge Functions → discord-bot → Logs');
}

runEnvTests().catch(console.error); 

File: fantasy-chess-frontend/test-discord-env.py
#!/usr/bin/env python3
"""
Test Discord Environment Variables
"""

import os
import requests
import json

# Supabase configuration
SUPABASE_URL = "https://wdbwzvnkfbyzazodfhsw.supabase.co"
SUPABASE_ANON_KEY = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6IndkYnd6dm5rZmJ5emF6b2RmaHN3Iiwicm9sZSI6ImFub24iLCJpYXQiOjE3NTM1NzU2MzIsImV4cCI6MjA2OTE1MTYzMn0.Ej8Ej8Ej8Ej8Ej8Ej8Ej8Ej8Ej8Ej8Ej8Ej8Ej8Ej8"

def test_discord_function():
    """Test the Discord function with detailed error reporting"""
    print("🧪 Testing Discord Function...")
    
    # Test data
    test_data = {
        "action": "create_league_channel",
        "leagueName": "Test League",
        "leagueId": "test-league-id"
    }
    
    try:
        # Call the Discord function
        response = requests.post(
            f"{SUPABASE_URL}/functions/v1/discord-bot",
            headers={
                "Authorization": f"Bearer {SUPABASE_ANON_KEY}",
                "Content-Type": "application/json"
            },
            json=test_data,
            timeout=30
        )
        
        print(f"Status Code: {response.status_code}")
        print(f"Response Headers: {dict(response.headers)}")
        print(f"Response Body: {response.text}")
        
        if response.status_code == 200:
            print("✅ Discord function is working!")
            return True
        else:
            print(f"❌ Discord function failed with status {response.status_code}")
            
            # Try to parse error response
            try:
                error_data = response.json()
                print(f"Error details: {json.dumps(error_data, indent=2)}")
            except:
                print("Could not parse error response as JSON")
            
            return False
            
    except requests.exceptions.Timeout:
        print("❌ Request timed out")
        return False
    except requests.exceptions.RequestException as e:
        print(f"❌ Request failed: {e}")
        return False
    except Exception as e:
        print(f"❌ Unexpected error: {e}")
        return False

def main():
    print("🔍 Discord Environment Test")
    print("=" * 40)
    
    # Test the function
    success = test_discord_function()
    
    print("\n" + "=" * 40)
    if success:
        print("🎉 Discord integration is working!")
    else:
        print("❌ Discord integration needs attention")
        print("\nCommon issues:")
        print("1. Discord bot token not set in Supabase")
        print("2. Discord main server ID not set in Supabase")
        print("3. Bot doesn't have proper permissions")
        print("4. Bot not added to the Discord server")
        print("\nCheck your Supabase project settings for:")
        print("- DISCORD_BOT_TOKEN")
        print("- DISCORD_MAIN_SERVER_ID")

if __name__ == "__main__":
    main() 